morf-oracle/src/main/java/org/alfasoftware/morf/jdbc/oracle/OracleMetaDataProvider.java

```diff
@@ -324,7 +324,7 @@ private void handleTableColumnRow(final Map<String, List<String>> primaryKeys,
 
     // -- Stage 3: find the index names...
     //
-    primaryKeyIndexNames = Sets.newTreeSet();
+    primaryKeyIndexNames = Maps.newHashMap();

```

Comment:
primaryKeyIndexNames is directly returned by primaryKeyIndexNames().
Which means you are returning your mutable HashMap to the Morf end users.
But this class is actually full of these mutable maps being given to the end user, so let's keep it as it is.
One day we will wrap them all in UnmodifiableMaps.. Unless we delete the dialect first :D

Reply:
Good point!